File: android/app/src/main/java/com/collectrndemo/modules/show/field/core/BaseShowView.java
package com.collectrndemo.modules.show.field.core;

import android.content.Context;
import android.content.res.Resources;

import androidx.annotation.NonNull;

import com.collectrndemo.modules.show.VGSShowOnCreateViewInstanceListener;
import com.collectrndemo.modules.util.ResourceUtil;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewGroupManager;
import com.facebook.react.uimanager.annotations.ReactProp;
import com.verygoodsecurity.vgsshow.widget.VGSTextView;

public abstract class BaseShowView extends ViewGroupManager<VGSTextView> {

    private VGSTextView view;

    private final VGSShowOnCreateViewInstanceListener listener;

    public BaseShowView(VGSShowOnCreateViewInstanceListener listener) {
        this.listener = listener;
    }

    protected abstract VGSTextView createView(@NonNull ThemedReactContext reactContext);

    @ReactProp(name = "contentPath")
    public void setContentPath(VGSTextView view, String text) {
        view.setContentPath(text);
    }

    @ReactProp(name = "padding")
    public void setPadding(VGSTextView view, int padding) {
        int paddingDp = ResourceUtil.convertPxToDp(view.getContext(), padding);
        view.setPadding(paddingDp, paddingDp, paddingDp, paddingDp);
    }

    @ReactProp(name = "fontSize")
    public void setFontSize(VGSTextView view, int size) {
        view.setTextSize(size * Resources.getSystem().getDisplayMetrics().density);
    }

    @ReactProp(name = "hint")
    public void setHint(VGSTextView view, String text) {
        view.setHint(text);
    }

    public String getContentPath() {
        return view == null ? null : view.getContentPath();
    }

    @NonNull
    @Override
    protected VGSTextView createViewInstance(@NonNull ThemedReactContext reactContext) {
        view = createView(reactContext);
        listener.onCreateViewInstance(view);
        return view;
    }

    protected int toDp(Context context, int px) {
        return ResourceUtil.convertPxToDp(context, px);
    }
}

File: android/app/src/main/kotlin/com/collectrndemo/simple/modules/show/field/core/BaseShowView.kt
package com.collectrndemo.simple.modules.show.field.core

import com.collectrndemo.simple.modules.OnCreateViewInstanceListener
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.ViewGroupManager
import com.verygoodsecurity.vgsshow.widget.VGSTextView

/**
 * @noinspection unused
 */
abstract class BaseShowView(
    private val listener: OnCreateViewInstanceListener<VGSTextView>
) : ViewGroupManager<VGSTextView?>() {

    private lateinit var view: VGSTextView

    protected abstract fun createView(reactContext: ThemedReactContext): VGSTextView

    override fun createViewInstance(reactContext: ThemedReactContext): VGSTextView {
        view = createView(reactContext)
        listener.onCreateViewInstance(view)
        return view
    }
}